components/ActionSheet/src/MDCActionSheetController.h

```diff
@@ -0,0 +1,179 @@
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+#ifndef MDC_SUBCLASSING_RESTRICTED
+#if defined(__has_attribute) && __has_attribute(objc_subclassing_restricted)
+#define MDC_SUBCLASSING_RESTRICTED __attribute__((objc_subclassing_restricted))
+#else
+#define MDC_SUBCLASSING_RESTRICTED
+#endif
+#endif  // #ifndef MDC_SUBCLASSING_RESTRICTED
+
+@class MDCActionSheetAction;
+
+/**
+ MDCActionSheetController displays an alert message to the user, similar to
+     UIAlertControllerStyleActionSheet.
+
+ A Material Action Sheet consist of a title, message and a list of actions.
+
+ Learn more at the [Material spec bottom
+ sheet](https://material.io/design/components/sheets-bottom.html)
+ [Material spec list](https://material.io/design/components/lists.html)
+
+ To learn more about
+ [UIAlertController](https://developer.apple.com/documentation/uikit/uialertcontroller)
+ or [UIAlertControllerSytleActionSheet](https://developer.apple.com/documentation/uikit/uialertcontrollerstyle/uialertcontrollerstyleactionsheet)
+
+ MDCActionSheetController does not support UIPopoverController, instead it will always be presented
+ in a sheet from the bottom.
+
+ ### Dependencies
+
+ MDCActionSheetController depends on BottomSheet Component.
+ */
+MDC_SUBCLASSING_RESTRICTED
+@interface MDCActionSheetController : UIViewController
+
+/**
+ Designated constructor to create and return a view controller for displaying an alert to the user.
+
+ After creating the alert controller, add actions to the controller by calling -addAction.
+
+ @param title The title of the alert.
+ @param message Descriptive text that summarizes a decision in a sentence or two.
+ @return An initialized MDCActionSheetController object.
+ */
++ (nonnull instancetype)actionSheetControllerWithTitle:(nullable NSString *)title
+                                               message:(nullable NSString *)message;
+
+/**
+ Convenience constructor to create and return a view controller for displaying an alert to the user.
+
+ After creating the alert controller, add actions to the controller by calling -addAction.
+
+ @param title The title of the alert.
+ @return An initialized MDCActionSheetController object.
+ */
++ (instancetype)actionSheetControllerWithTitle:(nullable NSString *)title;
+
+
+- (instancetype)init;
+
+/** Action sheet controllers must be created with actionSheetControllerWithTitle: or
+ with actionSheetControllerWithTitle:message:  */
+- (nonnull instancetype)initWithNibName:(NSString *)nibNameOrNil
+                                 bundle:(NSBundle *)nibBundleOrNil NS_UNAVAILABLE;
+
+/** Action sheet controllers must be created with actionSheetControllerwithTitle:
```

Comment:
_Style nit:_ We should be consistent (at least within a single file) about whether to put newlines after and before the "/**" and "*/" doc delimiter things